### PR TITLE
Nick & Joe's updates

### PR DESCRIPTION
**Pull request by nolds9**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,202 +1,55 @@
 # Objects and Functions
 
-## Learning Objectives (5 / 5)
+[![Build Status](https://travis-ci.org/ga-wdi-lessons/js-objects-functions.svg?branch=master)](https://travis-ci.org/ga-wdi-lessons/js-objects-functions)
```
Comment (jshawl) [inline readme.md:3]: failing

File: readme.md [hunk 2]
```diff
@@ -1,202 +1,55 @@
...
 
 ### Functions
 - Describe what a JavaScript function is.
 - Recognize the parts of a function.
 - Write a function in JavaScript using a declaration and an expression.
-- Define hoisting.
-- Differentiate between referencing and invoking a function.
 - State the difference between a function's output and side effects.
+- Differentiate between referencing and invoking a function.
+- Define hoisting.
```
Comment (jshawl) [inline readme.md:13]: :+1:

File: readme.md (hunk 1 above)
Comment (nolds9) [inline readme.md:3]: por que?

Comment (mattscilipoti) [inline readme.md:3]: The link references the "master" branch, rather than the branch for this PR.  I don't have an answer for managing that easily.

Comment (mattscilipoti) [inline readme.md:3]: If you click the link, you can see that it is passing for this branch.

File: readme.md [hunk 3]
```diff
@@ -1,40 +1,183 @@
...
+* Simply put, a function is a block of code that takes an input, process that input and then produces an output.
+* Analogy: Quizno's Oven
 
-Using an index.
+**Q. Think back to what we have covered so far, when have you used functions before?**
 
-## Objects (5 / 15)
+### Why do we use functions?
+<!--
+Say we wanted the square of a number without using the above function. How would we do that?
```
Comment (nolds9) [inline readme.md]: What above function?

Comment (joe-gz) [inline readme.md]: yea that was already in there. I commented it out so it wouldn't show, that's all!

File: readme.md [hunk 4]
```diff
@@ -1,40 +1,240 @@
...
 * Conditionals
+* Data Collections
+
+## Functions
+
+**What’s a function?**
+
+* Fundamental component of Javascript.
+* A reusable block of Javascript code.
+* Simply put, a function is a block of code that takes an input, process that input and then produces an output.
```
Comment (amaseda) [inline readme.md]: *processes that input

File: readme.md [hunk 5]
```diff
@@ -1,40 +1,240 @@
...
+**Function declarations** define functions without assigning them to variables.
+
+**Function expressions** save anonymous functions to variables.
+
+While we call/reference functions defined through declarations and expressions the same way, they do have a subtle but important difference...
+
+> **Note**: Declarations are processed before any code is executed, meaning you can call functions before they are declared. This behavior is known as **hoisting**.
+
+
+### Hoisting (10 / 40
```
Comment (amaseda) [inline readme.md]: *closing parentheses

File: readme.md [hunk 6]
```diff
@@ -1,40 +1,240 @@
...
+    var bar = function() {
+        return 3;
+    };
+    var bar = function() {
+        return 8;
+    };
+}
+alert(foo());
+```
+
```
Comment (amaseda) [inline readme.md:228]: :+1:

File: readme.md [hunk 7]
```diff
@@ -44,22 +244,21 @@ var car = {
...
-2. Menu items at a fast food restaurant
-3. Definitions in a dictionary
-4. List of students in WDI
-5. List of all GA courses ordered by city
+### You DO: Interacting with Objects (30 / 95)
 
-### Interacting with Objects (10 / 35)
+**Read through the below, and then complete the exercise with your partner**
 
 #### Create
```
Comment (amaseda) [inline readme.md:264]: On line 271, need a comma after `year: 1997` in the example car object.

**Comment by amaseda** [comment]
Left some notes. Overall :+1:

**Pull request merged**